### ISSUE TITLE
Does tf.keras.metrics.AUC work on multi-class problems?

### ISSUE THREAD

**Issue by ogreyesp** [tensorflow/addons#651]
Hi, 
I have a multi-class classification problem and I want to measure AUC on training and test data.

tf.keras has implemented AUC metric (tf.keras.metrics.AUC), but I'm not be able to see whether this metric could safely be used in multi-class problems. Even, the example "Classification on imbalanced data" on the official Web page is dedicated to a binary classification problem.

I have implemented a CNN model that predicts six classes, having a softmax layer that gives the probabilities of all the classes. I used this metric as follows

self.model.compile(loss='categorical_crossentropy',
                       optimizer=Adam(hp.get("learning_rate")),
                       metrics=['accuracy', AUC()]),

and the code was executed without any problem. However, sometimes I see some results that are quite strange for me. For example, the model reported an accuracy of 0.78333336 and AUC equal to 0.97327775, Is this possible? Can a model have a low accuracy and an AUC so high?

I wonder that, although the code does not give any error, the AUC metric is computing wrong.

Somebody may confirm me whether or not this metrics support multi-class classification problems?

Thanks in advance,

Oscar

**Comment by seanpmorgan** [comment]
Hi @ogreyesp! We're tracking the request for multi-class AUC in https://github.com/tensorflow/addons/issues/265

**Comment by YasarL** [comment]
i have a similar question: do tf.keras.metrics.Precision() and tf.keras.metrics.Recall() support mutliclass classification?  I do pixelwise semantic segmentation, so I was wondering if the values would be averaged pixelwise or categorywise ?

Or do tf.keras.metrics.Precision() / tf.keras.metrics.Recall() only support binary classification? If so, which class is the one that is used for calculating recall and precision?

**Comment by fmorenovr** [comment]
I have the similar behavior using AUC metric. 
Also, do you know how to load_model when you trained using AUC metrc?? I cannot do that.
